kategory-optics/src/main/kotlin/kategory/optics/Traversal.kt

```diff
@@ -0,0 +1,337 @@
+package kategory.optics
+
+import kategory.Applicative
+import kategory.Const
+import kategory.Either
+import kategory.HK
+import kategory.Id
+import kategory.IntMonoid
+import kategory.ListKW
+import kategory.Monoid
+import kategory.Option
+import kategory.applicative
+import kategory.identity
+import kategory.left
+import kategory.map
+import kategory.monoid
+import kategory.none
+import kategory.right
+import kategory.some
+import kategory.traverse
+import kategory.value
+
+/**
+ * [Traversal] is a type alias for [PTraversal] which fixes the type arguments
+ * and restricts the [PTraversal] to monomorphic updates.
+ */
+typealias Traversal<S, A> = PTraversal<S, S, A, A>
+
+/**
+ * A [Traversal] is an optic that allows to see into a structure with 0 to N targets.
+ *
+ * [Traversal] is a generalisation of [kategory.Traverse] and can be seen as a representation of modifyF.
+ * all methods are written in terms of modifyF
+ *
+ * @param S the source of a [PTraversal]
+ * @param T the modified source of a [PTraversal]
+ * @param A the target of a [PTraversal]
+ * @param B the modified target of a [PTraversal]
+ */
+abstract class PTraversal<S, T, A, B> {
+
+    abstract fun <F> modifyF(FA: Applicative<F>, s: S, f: (A) -> HK<F, B>): HK<F, T>
+
+    /**
+     * Modify polymorphically the target of a [PTraversal] with an Applicative function
+     */
+    inline fun <reified F> modifyF(s: S, crossinline f: (A) -> HK<F, B>): HK<F, T> = modifyF(applicative(), s, { b -> f(b) })
```

Comment:
Applicative should be a parameter with a default value, rather than being inlined. It's rarae, still we could have multiple instances of Applicative for an F.

Reply:
The reason for the inline overload here is because the modifyF method with an `Applicative<F>` param is abstract and thus cannot be inlined. So I overloaded it to provide a method that could be called with the global instance, I did the same for Fold (https://github.com/kategory/kategory/pull/308).

I am unsure if there is better solution or if we'd rather force the parameter instead of an overload like this.

Reply:
try this signature: `inline fun <reified F> modifyF(s: S, crossinline f: (A) -> HK<F, B>, AF: Applicative<F> = applicative())`